src/client/app/effects/snippet.ts

```diff
@@ -356,11 +362,21 @@ export class SnippetEffects {
             new Snippet.ImportSuccessAction(snippet)
         ];
 
+        let properties = {};
+        if (mode === Snippet.ImportType.GIST) {
+            properties = { ...properties, hashedGistId: sha1(snippet.gist)};
```

Comment:
I don't think properties needs to be immutable, you can just do properties['HashGistId'] = sha1(snippet.gist) it will be better performance.